src/lib/hash/hash.cpp

```diff
@@ -251,6 +255,18 @@ std::unique_ptr<HashFunction> HashFunction::create(std::string_view algo_spec,
       }
 #endif
 
+
+#if defined(BOTAN_HAS_KECCAK_FIPS)
+   if(req.algo_name() == "Keccak-FIPS[512]")
```

Comment:
As I said in the ticket, I really dislike this approach because the Keccak being exposed here is not, as I understand it, approved by NIST as a hash function, nor is it implemented by any library I am aware of.

Reply:
Agreed, that makes sense. I removed it from the external interface.